+дерево

ch3rnushka opened the pull request:
+таблица символов

некоторые тесты поменяла до 14:
-где конс нулл на пустой список
-где андефайнед переменные и функции
-и конд другой в файлике

ch3rnushka committed:
+дерево
+таблица символов

некоторые тесты поменяла до 14:
-где конс нулл на пустой список
-где андефайнед переменные и функции
-и конд другой в файлике

MattWay224 committed:
delete unnessesary comment

MattWay224 commented on src/steps/Parser.java:
```diff
@@ -241,40 +321,66 @@ private ASTNode parseOperation(String operator) throws Exception {
         return factory.createOperationNode(operator, operands, false, op.line);
     }
 
     private ASTNode parseFuncCall(String functionName) throws Exception {
```
на 14 тесте эта функция выкидывает ошибку из-за того, что функция создана в локальном месте, а не в глобальном

MattWay224 committed:
singleton pattern added
still need to fix 14-16 tests

MattWay224 committed:
Eval node

ch3rnushka committed:
+quote correct
+test 14 conds modify structure
due to
( cond Element1 Element2 [ Element3 ] )
The form is the construct for conditional evaluation. It contains two or three
arguments. The evaluation of the form starts from evaluating its first argument.
The result of the evaluation should be of type boolean. If the result is true the
second argument is evaluated, and the result becomes the result of the whole
form. Otherwise, the third argument is evaluated and the result becomes the
result of the whole form. If the result of the first argument is false and there is no
third argument in the form, the result of the whole form is null.

ch3rnushka committed:
test 17 no strings

MattWay224 reviewed (approved):
nice
